Tests/GRPCCoreTests/Transport/InProcessClientTransportTests.swift

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+
+@testable import GRPCCore
+
+final class InProcessClientTransportTests: XCTestCase {
+  func testConnectWhenConnected() async {
+    let retryPolicy = RetryPolicy(
+      maximumAttempts: 10,
+      initialBackoff: .seconds(1),
+      maximumBackoff: .seconds(1),
+      backoffMultiplier: 1.0,
+      retryableStatusCodes: [.unavailable]
+    )
+    let client = InProcessClientTransport(
+      server: .init(),
+      executionConfigurations: .init(defaultConfiguration: .init(retryPolicy: retryPolicy))
+    )
```

Comment:
This is in every test and adds quite a lot of noise, can we factor it out to a helper? `private func makeClient(configuration: Configuration? = nil, server: InProcessServerTransport = InProcessServerTransport())`